### PR TITLE
feature/dss-90-API_Modification for adding date and last modification…

### PR DESCRIPTION
**Pull request by FlorentDev64**
… on project

        [ch62224](https://app.clubhouse.io/dataiku/story/62224/api-to-get-project-creation-and-last-modification-date)

	Two new method/routes added to reflect changes of PublicAPIProjectController (GET /projects/{projectKey}/interest, /projects/{projectKey}/timeline?itemCount)

### CODE REVIEW

File: dataikuapi/dss/project.py [hunk 1]
```diff
@@ -249,6 +249,29 @@ def set_permissions(self, permissions):
...
+        :return: a dict object containing the project's interest
+        :rtype: dict
+        """
+        return self.client._perform_json("GET","/projects/%s/interest" % self.project_key)
+
+    def get_timeline(self, itemCount=100):
+        """
+        Get a timeline of this project. The timeline consists of information about the creation of this project
+        (by whom, and when), the last modification of this project (by whom and when), a list of contributors,
+        and a list of modification. This list contains "itemCount" number of element (default: 100). If itemCount
```
Comment (apichery) [inline dataikuapi/dss/project.py:265]: ...and a list of modification**s**.

Comment (apichery) [reply dataikuapi/dss/project.py:265]: >This list contains "itemCount" number of element (default: 100). If itemCount....

This list of modifications contains a maximum of itemCount elements (default: 100).

File: dataikuapi/dss/project.py [hunk 2]
```diff
@@ -249,6 +249,29 @@ def set_permissions(self, permissions):
         return self.client._perform_empty(
             "PUT", "/projects/%s/permissions" % self.project_key, body = permissions)
 
+    def get_interest(self):
+        """
+        Get the interest of this project. The interest means the number of watchers and the number of star.
+
+        :return: a dict object containing the project's interest
```
Comment (apichery) [inline dataikuapi/dss/project.py:256]: Might be useful to add the name of the 2 fields present the dict (starCount and watchCount).

File: dataikuapi/dss/project.py (hunk 1 above)
Comment (apichery) [reply dataikuapi/dss/project.py:265]: Also might be a good idea to describe the various top-level fields in the dict (createdOn, ...)

**Review by apichery — CHANGES_REQUESTED** [review]

File: dataikuapi/dss/project.py (hunk 1 above)
Comment (FlorentDev64) [reply dataikuapi/dss/project.py:265]: Should I describe them, or simply list them

**Review by apichery — APPROVED** [review]

**Pull request merged**